addons/base_vat/models/res_partner.py

```diff
@@ -251,6 +253,65 @@ def check_vat_ch(self, vat):
             return check == int(num[8])
         return False
 
+    def _checksum_ci_ec(self, number):
```

Comment:
Why would you need to copy all the validation code from `stdnum`? it's a dependency, you can call it. There are even handy shortcuts:
`stdnum.util.get_cc_module("ec", "ci")`
`stdnum.util.get_cc_module("ec", "ruc")`

PS: please never copy code without giving attribution, copyright and license, that's not nice (and it violates the original license, copyright law, and the Odoo CLA) ;)